ocp_resources/machineset.py

```diff
@@ -0,0 +1,189 @@
+            "type": "server",
+            "userDataSecret": {
+                "name": "worker-user-data"
+            }
+        }
+    }
+    """
+
+    api_group = NamespacedResource.ApiGroup.MACHINE_OPENSHIFT_IO
+
+    def __init__(
+        self,
+        cluster_name=None,
+        name=None,
+        namespace=None,
+        teardown=True,
+        client=None,
+        machine_role="worker",
+        machine_type="worker",
+        replicas=1,
+        provider_spec=None,
+        yaml_file=None,
+    ):
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            teardown=teardown,
+            yaml_file=yaml_file,
+        )
+        self.replicas = replicas
+        self.cluster_name = cluster_name
+        self.machine_role = machine_role
+        self.machine_type = machine_type
+        self.provider_spec = provider_spec or {}
+
+    @property
+    def available_replicas(self):
+        return self.instance.status.availableReplicas
+
+    @property
+    def ready_replicas(self):
+        return self.instance.status.readyReplicas
+
+    @property
+    def desired_replicas(self):
+        return self.instance.status.replicas
+
+    @property
+    def provider_spec_value(self):
+        return self.instance.spec.template.spec.providerSpec.value
+
+    def wait_for_replicas(self, timeout=TIMEOUT, sleep=1):
+        """
+        Wait for machine-set replicas to reach 'ready' state.
+
+        Args:
+            timeout (int): maximum time to wait_for_replicas for the 'ready' state.
+            sleep (int): sleep time between each sample.
+
+        Returns:
+            bool: True if machine-set reached 'ready' state, False otherwise.
+        """
+        try:
+            for ready_replicas in TimeoutSampler(
+                wait_timeout=timeout, sleep=sleep, func=lambda: self.ready_replicas
+            ):
+                if ready_replicas and ready_replicas == self.desired_replicas:
+                    return True
+        except TimeoutExpiredError:
+            LOGGER.error(
+                f"Machine-set {self.name} replicas failed to reach into 'ready' state, actual ready replicas: "
+                f"{self.ready_replicas}, desired replicas: {self.desired_replicas}"
+            )
+            return False
+
+    def scale_replicas(self, replicas, wait_timeout=TIMEOUT, sleep=1, wait=True):
+        """
+        Scale down/up a machine-set replicas.
+
+        Args:
+            replicas (int): num of replicas to scale_replicas to.
+            wait_timeout (int): maximum time to wait_for_replicas for scaling the machine-set.
+            sleep (int): sleep time between each sample of the machine-set state.
+            wait (bool): True if waiting for machine-set to reach into 'ready' state, False otherwise.
+
+        Returns:
+            bool: True if scaling the machine set was successful, False otherwise.
```

Comment:
Returns True also if wait=False (and not necessarily if scaling succeeded)

Reply:
I don't think its necessary to add it to the docstring.

basically if the update function didn't raise an exception, then that means  the update was successful.

Reply:
if wait=False, wait_for_replicas is not called.
A user should know from the docstring what is the expected behavior; for example:
`bool: True if scaling the machine set was successful or wait=False, False otherwise.`

Reply:
ack